model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1355,6 +1355,10 @@ message StandardRequirements {
     // This requirement indicates the requires_time_sorted_input field of ParDo
     // transform payloads must be inspected.
     REQUIRES_TIME_SORTED_INPUT = 3 [(beam_urn) = "beam:requirement:pardo:time_sorted_input:v1"];
+
+    // This requirement indicates the restriction_coder_id field of ParDo
+    // transform payloads must be inspected.
+    REQUIRES_SPLITTABLE_DOFN = 4 [(beam_urn) = "beam:requirement:pardo:splittable_dofn:v1"];
```

Comment:
Is this a requirement, given that the pipeline would still execute correctly were expansion not performed? (Maybe the performance is bad enough to merit this, but then again, maybe SDF is so primitive that all runners should do something do support sources.)

Reply:
If its unbounded, a runner will never be able to execute it without expansion which requires the runner to at least inspect whether its capable of doing it.

Reply:
Fair point.